utils/validate_meta_files.py

```diff
@@ -2,34 +2,56 @@
 
 
 def main():
-    asset_path = "Project/Assets"
+    asset_paths = ["Project/Assets", "com.unity.ml-agents"]
     meta_suffix = ".meta"
     python_suffix = ".py"
+    whitelist = frozenset(
+        [
+            "com.unity.ml-agents/.editorconfig",
+            "com.unity.ml-agents/.gitignore",
+            "com.unity.ml-agents/.npmignore",
+            "com.unity.ml-agents/Tests/.tests.json",
+        ]
+    )
+    ignored_dirs = {"Documentation~"}
 
     num_matched = 0
 
     unmatched = set()
 
-    for root, dirs, files in os.walk(asset_path):
-        dirs = set(dirs)
-        files = set(files)
-
-        combined = dirs | files
-        for f in combined:
-            if f.endswith(python_suffix):
-                # Probably this script; skip it
-                continue
-
-            # We expect each non-.meta file to have a .meta file, and each .meta file to have a non-.meta file
-            if f.endswith(meta_suffix):
-                expected = f.replace(meta_suffix, "")
-            else:
-                expected = f + meta_suffix
-
-            if expected not in combined:
-                unmatched.add(os.path.join(root, f))
-            else:
-                num_matched += 1
+    for asset_path in asset_paths:
+        for root, dirs, files in os.walk(asset_path, topdown=True):
+            # Modifying the dirs list with topdown=True will prevent us from recursing those directories
+            for ignored in ignored_dirs:
+                try:
+                    dirs.remove(ignored)
+                except ValueError:
+                    pass
+
+            dirs = set(dirs)
+            files = set(files)
+
+            combined = dirs | files
```

Comment:
that's cool

Reply:
Yeah, python sets have operator overloads. `&` gives the intersection, etc.
https://docs.python.org/3.7/library/stdtypes.html#set-types-set-frozenset